api/src/services/users.js

```diff
@@ -551,16 +551,21 @@ const validateUserContact = (data, user, userSettings) => {
   }
 };
 
-const createUserEntities = async (user, appUrl, preservePrimaryContact) => {
+const createUserEntities = async (data, appUrl) => {
+  // preserve the place's primary contact only if it's an existing one
```

Comment:
```suggestion
  // Preserve the place's primary contact, if data.place already exists in the DB. 
``` 
BTW. What happens if the people forget to add the place id? How often would an error like this happen?

Reply:
If the place id is not set it will create a new place and move the user's contact there which is the expected behavior, this is tested by unit tests in `api/tests/mocha/services/users.spec.js:1381`

This mistake can only happen when hitting the API directly with a script. The only place in the code base that calls this endpoint is in the admin app when creating or editing a user and the place property is always a string representing the place's id.

But re-reading some of the places-related code in the api, I will change the condition to check on the `_rev` property instead of `_id`, this is how `getOrCreatePlace()` decides if it should create a place or not